app/api/v1/mediator/ballot.py

```diff
@@ -230,17 +247,27 @@ def _get_election_parameters(
     if request_data.context:
         context = cast(CiphertextElectionContext, request_data.context)
     else:
-        context = CiphertextElectionContext.from_json_object(election.context)
+        context = election.context.to_sdk_format()
 
     return manifest, context, election_id
 
 
 def _submit_ballots(
     election_id: str, ballots: List[SubmittedBallot], settings: Settings = Settings()
 ) -> BaseResponse:
+    logger.info("submitting ballots")
     set_response = set_ballots(election_id, ballots, settings)
     if set_response.is_success():
+        logger.info(f"successfully set ballots: {str(set_response)}")
         inventory = get_ballot_inventory(election_id, settings)
+        if inventory is None:
+            inventory = BallotInventory(
```

Comment:
It feels wrong to write this code.  I don't understand how submitting ballots ever worked, but for the very first time anyone ever uploads a ballot that ballot absolutely needs to be created, and I couldn't find anywhere else that might possibly happen, so afaik this code is necessary.